kolibri/core/content/utils/upgrade.py

```diff
@@ -109,26 +109,26 @@ def diff_stats(channel_id, method, drive_id=None, baseurl=None):
 
 def count_new_resources_available_for_import(destination, channel_id):
     """
-    Queries the destination db to get the count of leaf nodes.
-    Subtract by the count of leaf nodes on default db to get the number of new resources.
+    Queries the destination db to get leaf nodes.
+    Subtract total number of leaf nodes by the count of leaf nodes on default db to get the number of new resources.
     """
     bridge = Bridge(app_name=CONTENT_APP_NAME, sqlite_file_path=destination)
     ContentNodeClass = bridge.get_class(ContentNode)
-    leaf_node_counts = (
-        bridge.session.query(ContentNodeClass.id)
+    leaf_node_ids = [
+        i
+        for i, in bridge.session.query(ContentNodeClass.id)
         .filter(
             ContentNodeClass.channel_id == channel_id,
             ContentNodeClass.kind != content_kinds.TOPIC,
         )
+        .all()
+    ]
+    return (
+        len(leaf_node_ids)
+        - ContentNode.objects.filter_by_uuids(leaf_node_ids, validate=False)
+        .filter(channel_id=channel_id)
         .count()
     )
```

Comment:
Seems like this might be unnecessarily expensive, pulling the list if IDs into local memory?

Would this be cheaper like

```python
    leaf_node_count = bridge.session.query(ContentNodeClass.id)
        .filter(
            ContentNodeClass.channel_id == channel_id,
            ContentNodeClass.kind != content_kinds.TOPIC,
        )
        .count()
    return (
        leaf_node_count
        - ContentNode.objects.filter_by_uuids(leaf_node_ids, validate=False)
        .filter(channel_id=channel_id)
        .count()
```

Reply:
But then `leaf_node_ids` is not defined  - this is needed to cross-reference the ids in the external DB with the ids in the internal DB.